python/cudf/cudf/core/index.py

```diff
@@ -383,7 +383,14 @@ def to_series(self):
 
         return Series(self._values)
 
-    @property
+    def __array__(self, dtype=None):
```

Comment:
Can we not just define `__array__` once, in `Frame` rather than duplicating in each of its subclasses?

Reply:
Yep, we can get past duplication(removed in Series & Index), but will still need it separately in `DataFrame` and `StringColumn` because of the different error messages we want to display.